test/functional/feature_anchors.py

```diff
@@ -63,17 +63,25 @@ def run_test(self):
         self.log.info("Check the addresses in anchors.dat")
 
         with open(node_anchors_path, "rb") as file_handler:
-            anchors = file_handler.read().hex()
+            anchors = file_handler.read()
 
+        anchors_hex = anchors.hex()
         for port in block_relay_nodes_port:
             ip_port = ip + port
-            assert ip_port in anchors
+            assert ip_port in anchors_hex
         for port in inbound_nodes_port:
             ip_port = ip + port
-            assert ip_port not in anchors
+            assert ip_port not in anchors_hex
 
-        self.log.info("Start node")
-        self.start_node(0)
+        self.log.info("Perturb anchors.dat to test it doesn't throw an error during initialization")
+        with self.nodes[0].assert_debug_log(["0 block-relay-only anchors will be tried for connections."]):
+            with open(node_anchors_path, "wb") as out_file_handler:
+                tweaked_contents = bytearray(anchors)
+                tweaked_contents[20:20] = b'1'
```

Comment:
how do you know that the existing byte is not `b'1'` and this is not a no-op. Even if it isn't a no-op, this may still pass de-serialization?

Reply:
I had the same question in my mind. I tested it manually, but the way I found to know it didn't pass de-serialization is checking the debug log: "0 block-relay-only anchors will be tried for connections.". If it had passed, it would try to connect to the 2 anchors from the file.

Reply:
Any other thoughts?